server-phoenix/lib/helios_web/schema/resolvers/event_collection.ex

```diff
@@ -24,6 +24,8 @@ defmodule HeliosWeb.Schema.Resolvers.EventCollection do
     {:ok,
      %{
        all: Repo.all(event_query),
+       by_author:
+         event_query |> Event.top_commits_by_author(Map.get(args, :top, 3)) |> Repo.all(),
        count: Event.count(Event) |> Repo.all() |> Enum.into(%{})
```

Comment:
What if we refactored this to:
* return `{:ok, query}`
* name a sub-schema to represent the result of events. `EventCollectionAggregation`?
* extract `all` and `count` as their own fields with resolvers into `EventCollectionAggregation`. The first argument to the resolver is `parent` and should be the resolution of the parent object, in this case `{:ok, query}`

Then you could add `by_author` as a field with its own arguments and resolver.